workitem/workitem_repository_blackbox_test.go

```diff
@@ -107,19 +95,48 @@ func (s *workItemRepoBlackBoxTest) TestSaveForUnchangedCreatedDate() {
 	defer gormsupport.DeleteCreatedEntities(s.DB)()
 
 	wi, err := s.repo.Create(
-		context.Background(), "system.bug",
+		context.Background(), workitem.SystemBug,
 		map[string]interface{}{
 			workitem.SystemTitle: "Title",
 			workitem.SystemState: workitem.SystemStateNew,
 		}, "xx")
-
-	if err != nil {
-		s.T().Error("Could not create workitem", err)
-	}
+	require.Nil(s.T(), err, "Could not create work item")
 
 	wi, err = s.repo.Load(context.Background(), wi.ID)
 
 	wiNew, err := s.repo.Save(context.Background(), *wi)
 
 	assert.Equal(s.T(), wi.Fields[workitem.SystemCreatedAt], wiNew.Fields[workitem.SystemCreatedAt])
 }
+
+func (s *workItemRepoBlackBoxTest) TestCreateWorkItemWithDescriptionNoMarkup() {
+	defer gormsupport.DeleteCreatedEntities(s.DB)()
+
+	wi, err := s.repo.Create(
+		context.Background(), workitem.SystemBug,
+		map[string]interface{}{
+			workitem.SystemTitle:       "Title",
+			workitem.SystemDescription: workitem.MarkupContent{Content: "Description"},
+			workitem.SystemState:       workitem.SystemStateNew,
+		}, "xx")
+	require.Nil(s.T(), err, "Could not create work item")
+
+	wi, err = s.repo.Load(context.Background(), wi.ID)
+	// app.WorkItem does not contain the markup associated with the description (yet)
+	assert.Equal(s.T(), workitem.MarkupContent{Content: "Description", Markup: workitem.SystemMarkupDefault}, wi.Fields[workitem.SystemDescription])
+}
+
+func (s *workItemRepoBlackBoxTest) TestCreateWorkItemWithDescriptionMarkup() {
+	defer gormsupport.DeleteCreatedEntities(s.DB)()
+	wi, err := s.repo.Create(
+		context.Background(), workitem.SystemBug,
+		map[string]interface{}{
+			workitem.SystemTitle:       "Title",
+			workitem.SystemDescription: workitem.MarkupContent{Content: "Description", Markup: workitem.SystemMarkupAsciidoc},
+			workitem.SystemState:       workitem.SystemStateNew,
+		}, "xx")
+	require.Nil(s.T(), err, "Could not create work item")
+	wi, err = s.repo.Load(context.Background(), wi.ID)
+	// app.WorkItem does not contain the markup associated with the description (yet)
+	assert.Equal(s.T(), workitem.MarkupContent{Content: "Description", Markup: workitem.SystemMarkupAsciidoc}, wi.Fields[workitem.SystemDescription])
```

Comment:
ok, I understand your concerns about refactoring, but I think that we loose in clarity: I really like the way 'go' supports named arguments in function calls. Anyway, I'll follow your guidelines.